python/src/filesystem.rs

```diff
@@ -531,10 +531,10 @@ impl ObjectOutputStream {
         Err(PyNotImplementedError::new_err("'read' not implemented"))
     }
 
-    fn write(&mut self, data: Vec<u8>) -> PyResult<i64> {
+    fn write(&mut self, data: &PyBytes) -> PyResult<i64> {
```

Comment:
Using `Vec<u8>` meant PyO3 was cloning all the input data. By using `PyBytes` we avoid the copy.